sda-commons-server-auth/src/test/java/org/sdase/commons/server/auth/key/RsaKeyLoaderSchedulerTest.java

```diff
@@ -0,0 +1,128 @@
+package org.sdase.commons.server.auth.key;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.mockito.Mockito.mock;
+
+import java.security.interfaces.RSAPublicKey;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class RsaKeyLoaderSchedulerTest {
+  private static final int RELOAD_PERIOD = 1;
+
+  // test data
+  private final List<LoadedPublicKey> actualKeys = new ArrayList<>();
+  private final AtomicBoolean actualInvocationFails = new AtomicBoolean();
+
+  // track invocations
+  private final AtomicInteger numberOfReloads = new AtomicInteger(0);
+
+  // setup
+  private final RsaPublicKeyLoader keyLoader = new RsaPublicKeyLoader();
+  private final ScheduledExecutorService executorService =
+      Executors.newSingleThreadScheduledExecutor();
+  private final RsaKeyLoaderScheduler keyLoaderScheduler =
+      RsaKeyLoaderScheduler.create(keyLoader, executorService);
+  private final KeySource keySource =
+      () -> {
+        numberOfReloads.incrementAndGet();
+        if (actualInvocationFails.get()) {
+          throw new RuntimeException("Some error occurred during reload");
+        }
+        return actualKeys;
+      };
+
+  @Before
+  public void startKeyLoader() {
+    keyLoader.addKeySource(keySource);
+    keyLoaderScheduler.internalStart(0, RELOAD_PERIOD, TimeUnit.SECONDS);
+    // wait for initial load
+    waitForNextReload();
+  }
+
+  @After
+  public void shutdownExecutorService() {
+    executorService.shutdown();
+  }
+
+  @Test
+  public void shouldNotBeAuthenticatedWithRevokedKey() {
```

Comment:
Since we don't authenticate anything but just check if the keys are there, maybe:

```suggestion
  public void shouldRemoveRevokedKey() {
```